File: code/src/types/component.ts
/**
 * 元器件相关类型定义
 */

/** 验证状态 */
export const VerificationStatus = {
  Unverified: 'unverified',
  Verifying: 'verifying',
  Verified: 'verified',
  NotRecommended: 'not_recommended',
} as const;
export type VerificationStatus = (typeof VerificationStatus)[keyof typeof VerificationStatus];

/** 元器件 */
export interface Component {
  id: string;
  partNumber: string;
  manufacturer: string;
  category: string;
  subCategory: string;
  description: string;
  packageType: string;
  specifications: string;
  datasheetUrl: string;
  defaultUnit: string;
  isActive: boolean;
  hasAlternatives?: boolean;
  alternativeCount?: number;
  usedInBoms?: number;
  createdAt: string;
  updatedAt: string;
}

/** 创建元器件参数 */
export interface CreateComponentParams {
  partNumber: string;
  manufacturer?: string;
  category?: string;
  subCategory?: string;
  description?: string;
  packageType?: string;
  specifications?: Record<string, unknown>;
  datasheetUrl?: string;
  defaultUnit?: string;
}

/** 更新元器件参数 */
export interface UpdateComponentParams {
  componentId: string;
  partNumber?: string;
  manufacturer?: string;
  category?: string;
  description?: string;
  packageType?: string;
  specifications?: Record<string, unknown>;
  datasheetUrl?: string;
}

/** 替代料 */
export interface AlternativePart {
  id: string;
  originalComponentId: string;
  alternativeComponentId: string;
  alternativeComponent?: Component;
  priority: number;
  verificationStatus: VerificationStatus;
  notes: string;
  verifiedBy: string;
  verifiedAt: string | null;
  createdAt: string;
  updatedAt: string;
}

/** 添加替代料参数 */
export interface AddAlternativeParams {
  originalComponentId: string;
  alternativeComponentId: string;
  priority?: number;
  verificationStatus?: VerificationStatus;
  notes?: string;
}

/** 更新替代料参数 */
export interface UpdateAlternativeParams {
  alternativeId: string;
  priority?: number;
  verificationStatus?: VerificationStatus;
  notes?: string;
  verifiedBy?: string;
}

/** 元器件搜索参数 */
export interface ComponentSearchParams {
  keyword?: string;
  category?: string;
  manufacturer?: string;
  hasAlternatives?: boolean;
  page?: number;
  pageSize?: number;
}

/** 供应商 */
export interface Supplier {
  id: string;
  name: string;
  contactPerson: string;
  contactEmail: string;
  contactPhone: string;
  website: string;
  leadTimeDays: number | null;
  notes: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

/** 元器件供应商关联 */
export interface ComponentSupplier {
  id: string;
  componentId: string;
  supplierId: string;
  supplier?: Supplier;
  supplierPartNumber: string;
  unitPrice: number | null;
  currency: string;
  moq: number | null;
  leadTimeDays: number | null;
  isPreferred: boolean;
  createdAt: string;
  updatedAt: string;
}

File: code/src-tauri/src/models/component.rs
use serde::{Deserialize, Serialize};

/// 元器件
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Component {
    pub id: String,
    pub part_number: String,
    pub manufacturer: String,
    pub category: String,
    pub sub_category: String,
    pub description: String,
    pub package_type: String,
    pub specifications: String,
    pub datasheet_url: String,
    pub default_unit: String,
    pub is_active: bool,
    pub has_alternatives: Option<bool>,
    pub alternative_count: Option<i64>,
    pub used_in_boms: Option<i64>,
    pub created_at: String,
    pub updated_at: String,
}

/// 创建元器件请求
#[derive(Debug, Deserialize)]
pub struct CreateComponentRequest {
    pub part_number: String,
    #[serde(default)]
    pub manufacturer: String,
    #[serde(default)]
    pub category: String,
    #[serde(default)]
    pub sub_category: String,
    #[serde(default)]
    pub description: String,
    #[serde(default)]
    pub package_type: String,
    #[serde(default)]
    pub datasheet_url: String,
    #[serde(default = "default_unit")]
    pub default_unit: String,
}

/// 更新元器件请求
#[derive(Debug, Deserialize)]
pub struct UpdateComponentRequest {
    pub component_id: String,
    pub part_number: Option<String>,
    pub manufacturer: Option<String>,
    pub category: Option<String>,
    pub description: Option<String>,
    pub package_type: Option<String>,
    pub datasheet_url: Option<String>,
}

/// 替代料
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct AlternativePart {
    pub id: String,
    pub original_component_id: String,
    pub alternative_component_id: String,
    pub alternative_component: Option<Component>,
    pub priority: i32,
    pub verification_status: String,
    pub notes: String,
    pub verified_by: Option<String>,
    pub verified_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

/// 添加替代料请求
#[derive(Debug, Deserialize)]
pub struct AddAlternativeRequest {
    pub original_component_id: String,
    pub alternative_component_id: String,
    #[serde(default = "default_priority")]
    pub priority: i32,
    #[serde(default = "default_verification_status")]
    pub verification_status: String,
    #[serde(default)]
    pub notes: String,
}

/// 更新替代料请求
#[derive(Debug, Deserialize)]
pub struct UpdateAlternativeRequest {
    pub alternative_id: String,
    pub priority: Option<i32>,
    pub verification_status: Option<String>,
    pub notes: Option<String>,
    pub verified_by: Option<String>,
}

/// 元器件搜索参数
#[derive(Debug, Deserialize)]
pub struct ComponentSearchParams {
    pub keyword: Option<String>,
    pub category: Option<String>,
    pub manufacturer: Option<String>,
    pub has_alternatives: Option<bool>,
    #[serde(default = "default_page")]
    pub page: i64,
    #[serde(default = "default_page_size")]
    pub page_size: i64,
}

fn default_unit() -> String { "PCS".to_string() }
fn default_priority() -> i32 { 1 }
fn default_verification_status() -> String { "unverified".to_string() }
fn default_page() -> i64 { 1 }
fn default_page_size() -> i64 { 20 }